### DISCUSSION TITLE
help with generating a token via link using the app router

### DISCUSSION THREAD

**Discussion by imraneuf** [vercel/next.js#63029] [Help]
### Summary

HI, i'm trying to export my project from using pages to app router. i having problem when generating jwt token when clicking a link  when using the app router. if using nextjs pages, it does work. the suggestion given is to use useRouter from next/navigation but tried that and ist not working. this the code i used  from browser console it shows error 404 at the line  const response = await axios.post('/api/generateToken');.  thank you for the help

"use client"

import { useRouter } from 'next/navigation'
import axios from 'axios';

export default function Home() {
  const router = useRouter()

  const handleGenerateToken = async () => {
    try {
      const response = await axios.post('/api/generateToken');
      const data = response.data;
      const { token } = data;
      alert(`Generated Token: ${token}`);
      const serverUrl = `http://localhost/validate.php?token=${token}`;
      window.location.href = serverUrl;
    } catch (error) {
      console.error(error);
    }
  }

  return (
    <div>
      <a href="#" onClick={handleGenerateToken}>Generate Token and Navigate</a>
    </div>
  )
}

**Comment by icyJoseph** [comment]
Hi,

Do you have `app/api/generateToken/route.ts`? in pages route it'd have been:

- `app/api/generateToken.ts`
- But in App Router, the handlers use: `app/api/generateToken/route.ts` 

Or `.js` if you are doing JS only.

**Reply by imraneuf** [reply]
hi thanks for the reply, let me check on the route.ts

**Reply by icyJoseph** [reply]
If you do have a route file, make sure to have the right name for the methods, GET, POST, etc
